packages/client/lib/rpc/modules/engine.ts

```diff
@@ -628,11 +629,15 @@ export class Engine {
     if (payloadAttributes) {
       const { timestamp, prevRandao, suggestedFeeRecipient } = payloadAttributes
       const parentBlock = this.chain.blocks.latest!
-      const payloadId = await this.pendingBlock.start(await this.vm.copy(), parentBlock, {
-        timestamp,
-        mixHash: prevRandao,
-        coinbase: suggestedFeeRecipient,
-      })
+      const payloadId = await this.pendingBlock.start(
+        (await this.vm.copy()) as VM<EVM, EEI>,
```

Comment:
Does this need to be cast?

Reply:
Since the `start` method only works with `EVM` and `EEI` (and not the generic types) we have to either typecast it here, or add generics to the start method, which is a massive rabbit hole as it basically requires adding generics in dozens of instances of the client package. That seemed overkill so I went for the typecast .